Current (5.0.7) installation has OpenSearch problems

OpenSearch fails to start:
```
Jan 25 21:54:44 scratch systemd-entrypoint[1610]: Invalid maximum heap size: -Xmx0g
```
In `/etc/opensearch/jvm.options`, changing `-Xmx0g` to `-Xmx1g` makes it start, but...

```
Jan 25 21:56:17 scratch systemd-entrypoint[2660]: WARNING: A terminally deprecated method in java.lang.System has been called
Jan 25 21:56:17 scratch systemd-entrypoint[2660]: WARNING: System::setSecurityManager has been called by org.opensearch.bootstrap.Op
Jan 25 21:56:17 scratch systemd-entrypoint[2660]: WARNING: Please consider reporting this to the maintainers of org.opensearch.boots
Jan 25 21:56:17 scratch systemd-entrypoint[2660]: WARNING: System::setSecurityManager will be removed in a future release
```

And…
```
# curl -k https://localhost:9200
OpenSearch Security not initialized.
```

Also seeing some errors during install (sorry, this is a bit of a dump):
```
Created symlink from /etc/systemd/system/multi-user.target.wants/opensearch.service to /usr/lib/systemd/system/opensearch.service.
Created symlink from /etc/systemd/system/multi-user.target.wants/logstash.service to /etc/systemd/system/logstash.service.
Generating RSA private key, 2048 bit long modulus
...........................................+++
.................................................................................................................................................................+++
e is 65537 (0x10001)
Generating RSA private key, 2048 bit long modulus
........+++
...................................+++
e is 65537 (0x10001)
Signature ok
subject=/O=perfSONAR/OU=Archive/CN=admin
Getting CA Private Key
Generating RSA private key, 2048 bit long modulus
...................................................+++
...+++
e is 65537 (0x10001)
Signature ok
subject=/O=perfSONAR/OU=Archive/CN=localhost
Getting CA Private Key
Certificate was added to keystore
[Generating opensearch passwords]
[DONE]

Adding password for user perfsonar
tee: /etc/perfsonar/logstash/proxy_auth.json: No such file or directory
[Creating pscheduler users]
[Creating pscheduler_logstash user]
[Creating pscheduler_reader user]
[Creating pscheduler_writer user]
[DONE]

[Creating pscheduler roles]
[Creating pscheduler_logstash role]
[Creating pscheduler_reader role]
[Creating pscheduler_writer role]
[Creating opendistro_security_anonymous role]
[DONE]

[Mapping pscheduler users to pscheduler roles]
[Mapping pscheduler_logstash user to pscheduler_logstash role]
[Mapping pscheduler_reader user to pscheduler_reader role]
[Mapping pscheduler_writer user to pscheduler_writer role]
[Mapping anonymous user to anonymous user backend role]
[DONE]

[Configure logstash]
grep: /etc/sysconfig/logstash: No such file or directory
[DONE]

Job for opensearch.service failed because the control process exited with error code. See "systemctl status opensearch.service" and "journalctl -xe" for details.
**************************************************************************
** This tool will be deprecated in the next major release of OpenSearch **
** https://github.com/opensearch-project/security/issues/1755           **
**************************************************************************
Security Admin v7
Will connect to localhost:9200
ERR: Seems there is no OpenSearch running on localhost:9200 - Will exit
Waiting for opensearch API to start...
[ERROR] API start timeout
warning: %post(perfsonar-archive-5.0.7-1.el7.noarch) scriptlet failed, exit status 1
Non-fatal POSTIN scriptlet failure in rpm package perfsonar-archive-5.0.7-1.el7.noarch
```

--- a/perfsonar-archive/perfsonar-archive/opensearch-scripts/pselastic_secure_pre.sh
+++ b/perfsonar-archive/perfsonar-archive/opensearch-scripts/pselastic_secure_pre.sh
@@ -116,12 +116,9 @@ echo -e '\n######## End OpenSearch Security PerfSONAR Configuration ########' | 
 
 ## Specify initial and maximum JVM heap sizes.
 
-HALF_MEM=$(free --giga | grep Mem: | awk '{ print int($2/2) }')
-if [ "$HALF_MEM" -gt "8" ]; then
-    HALF_MEM="8"
-fi
-sed -i "s/^-Xms.*/-Xms${HALF_MEM}g/g" $JVM_FILE
-sed -i "s/^-Xmx.*/-Xmx${HALF_MEM}g/g" $JVM_FILE
+HALF_MEM=$(free --mega | awk '$1 == "Mem:" { half=int(($2/2)+0.5); print (half > 8192) ? 8192 : half }')
+sed -i "s/^-Xms.*/-Xms${HALF_MEM}m/g" $JVM_FILE
+sed -i "s/^-Xmx.*/-Xmx${HALF_MEM}m/g" $JVM_FILE
 
 # Give execute permission to opensearch hash script
 chmod +x ${OPENSEARCH_SECURITY_PLUGIN}/tools/hash.sh

